src/equations/shallow_water_multilayer_1d.jl

```diff
@@ -353,6 +376,79 @@ In the two-layer setting this combination is equivalent to the fluxes in:
     return SVector(f)
 end
 
+"""
+    hydrostatic_reconstruction_ersing_etal(u_ll, u_rr, equations::ShallowWaterMultiLayerEquations1D)
+
+!!! warning "Experimental code"
+    This is an experimental feature and may change in future releases.
+
+A particular type of hydrostatic reconstruction of the water height and bottom topography to 
+guarantee well-balancedness in the presence of wet/dry transitions and entropy stability for the 
+[`ShallowWaterMultiLayerEquations1D`](@ref). 
+The reconstructed solution states `u_ll_star` and `u_rr_star` variables are used to evaluate the 
+surface numerical flux at the interface. The key idea is a piecewise linear reconstruction of the 
+bottom topography and water height interfaces using subcells, where the bottom topography is allowed 
+to be discontinuous. 
+Use in combination with the generic numerical flux routine [`Trixi.FluxHydrostaticReconstruction`](@extref).
+"""
+@inline function hydrostatic_reconstruction_ersing_etal(u_ll, u_rr,
+                                                        equations::ShallowWaterMultiLayerEquations1D)
+    # Unpack waterheight and bottom topographies
+    h_ll = waterheight(u_ll, equations)
+    h_rr = waterheight(u_rr, equations)
+    b_ll = u_ll[end]
+    b_rr = u_rr[end]
+
+    # Get the velocities on either side
+    v_ll = MVector(velocity(u_ll, equations))
+    v_rr = MVector(velocity(u_rr, equations))
+
+    threshold = equations.threshold_limiter
+
+    # Ensure zero velocity at dry states
+    for i in eachlayer(equations)
+        if h_ll[i] <= threshold
+            v_ll[i] = zero(eltype(u_ll))
+        end
+        if h_rr[i] <= threshold
+            v_rr[i] = zero(eltype(u_rr))
+        end
+    end
+
+    # Calculate interface heights
+    H_ll = waterheight(cons2prim(u_ll, equations), equations)
+    H_rr = waterheight(cons2prim(u_rr, equations), equations)
+
+    # Discontinuous reconstruction of the bottom topography
+    b_ll_star = min(H_ll[1], max(b_rr, b_ll))
+    b_rr_star = min(H_rr[1], max(b_rr, b_ll))
+
+    # Calculate reconstructed interface heights
```

Comment:
```suggestion
    # Calculate reconstructed interface total water heights
```
Or now that I write this again maybe "total layer heights"

Reply:
I would prefer to rename it as "total layer height".
"Total water height" makes sense for the SWE but with multiple layers the meaning is a bit unclear.

Reply:
I have renamed "interface heights" -> "total water heights" in 3d6d89a9381cd544f4ee470550c38ee47df42bf1